### DISCUSSION TITLE
Route-level cache fails when Granian's `loop_opt` is set to `True`.

### DISCUSSION THREAD

**Discussion by winstxnhdw** [litestar-org/litestar#3674] [Potential Issues]
Doubt this will be fixed but just a heads up for others.

**Comment by provinzkraut** [comment]
How does it fail?

**Reply by winstxnhdw** [reply]
Litestar will throw an exception from sniffio that it is unable to identify the current event loop. This is probably because Granian uses its own event loop from Rust.

**Reply by provinzkraut** [reply]
Well that's not an issue specific to the caching then. It's just that Granian's event loop isn't a drop-in replacement, or more specifically, that it's incompatible with [`anyio`](https://github.com/agronholm/anyio/).

**Reply by winstxnhdw** [reply]
Strange because using get_event_loop from the route works with no problem.

**Reply by provinzkraut** [reply]
Sure. It's probably *mostly* compatible, but in this particular instance, it's not.

There's not a lot we can do here. If you want support for this, the place to ask would be over at granian or anyio, since they are the ones that'd have to implement it.

**Reply by winstxnhdw** [reply]
Yeap, I figured. I doubt Granian can do anything about this either. I’ll just have loop_opt disabled.
